webapp/src/main/java/com/box/l10n/mojito/service/translationkit/TranslationKitService.java

```diff
@@ -396,4 +404,44 @@ public void checkForPartiallyImported(Long dropId) {
 
         return textUnitDTOs;
     }
+
+    private boolean isEmptyForStatus(Long repositoryId, Long localeId, StatusFilter statusFilter) {
+        TextUnitAndWordCount textUnitAndWordCount;
```

Comment:
that seems to be doing extra queries and potentially may get out of sync the TranslationKitStep logic. 

TranslationKitStep computes the number of words (`wordCount` attribute) that that can be used to check for empty kit. It is also saved in the translationKit DB entry. I would rely one the already existing word count and add it to TranslationKitAsXliff.

Not sure what is the best between re-reading from the DB entity or getting from the step (or potentially use an annotation). It is pretty similar.